packages/jest-types/src/TestPathPatterns.ts

```diff
@@ -5,42 +5,83 @@
  * LICENSE file in the root directory of this source tree.
  */
 
-import type {Config} from '@jest/types';
 import {escapePathForRegex, replacePathSepForRegex} from 'jest-regex-util';
 
-type PatternsConfig = {
+export class TestPathPatterns {
```

Comment:
didn't notice this was here before 😅 

I don't wanna export non-types from this package. I'm fine with not having it in `jest-util`, but could you pull it out into its own new package?